pkg/handlers/handlers.go

```diff
@@ -117,18 +142,29 @@ func newFileReader(r io.Reader) (fileReader, error) {
 
 	// Reset the reader to the beginning again to allow the handler to read from the start.
 	// This is necessary because Identify consumes the reader.
-	if _, err := fReader.Seek(0, io.SeekStart); err != nil {
+	if _, err = fReader.Seek(0, io.SeekStart); err != nil {
 		return fReader, fmt.Errorf("error resetting reader after archive identification: %w", err)
 	}
 
 	return fReader, nil
 }
 
+// DataOrErr represents a result that can either contain data or an error.
+// The Data field holds the byte slice of data, and the Err field holds any error that occurred.
+// This structure is used to handle asynchronous file processing where each chunk of data
+// or potential error needs to be communicated back to the caller. It allows for
+// efficient streaming of file contents while also providing a way to propagate errors
+// that may occur during the file handling process.
+type DataOrErr struct {
+	Data []byte
+	Err  error
+}
```

Comment:
Just noticing that `DataOrErr` pretty much describes Rust's [Result](https://doc.rust-lang.org/std/result/) enum. There are some useful [methods](https://doc.rust-lang.org/std/result/enum.Result.html#implementations) if you wanted to follow their pattern.

Reply:
🦀 🦀 🦀